java/ql/src/semmle/code/java/frameworks/apache/Collections.qll

```diff
@@ -262,3 +436,531 @@ private class ApacheMapUtilsModel extends SummaryModelCsv {
+        "org.apache.commons.collections4;MultiSetUtils;true;synchronizedMultiSet;;;Element of Argument[0];Element of ReturnValue;value",
+        "org.apache.commons.collections4;MultiSetUtils;true;unmodifiableMultiSet;;;Element of Argument[0];Element of ReturnValue;value",
+        "org.apache.commons.collections;MultiSetUtils;true;predicatedMultiSet;;;Element of Argument[0];Element of ReturnValue;value",
+        "org.apache.commons.collections;MultiSetUtils;true;synchronizedMultiSet;;;Element of Argument[0];Element of ReturnValue;value",
+        "org.apache.commons.collections;MultiSetUtils;true;unmodifiableMultiSet;;;Element of Argument[0];Element of ReturnValue;value"
+      ]
+  }
+}
+
+/**
+ * Value-propagating models for the class `org.apache.commons.collections4.QueueUtils`.
+ */
+private class ApacheQueueUtilsModel extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        "org.apache.commons.collections4;QueueUtils;true;predicatedQueue;;;Element of Argument[0];Element of ReturnValue;value",
+        "org.apache.commons.collections4;QueueUtils;true;synchronizedQueue;;;Element of Argument[0];Element of ReturnValue;value",
+        "org.apache.commons.collections4;QueueUtils;true;transformingQueue;;;Element of Argument[0];Element of ReturnValue;value",
+        "org.apache.commons.collections4;QueueUtils;true;unmodifiableQueue;;;Element of Argument[0];Element of ReturnValue;value",
+        "org.apache.commons.collections;QueueUtils;true;predicatedQueue;;;Element of Argument[0];Element of ReturnValue;value",
+        "org.apache.commons.collections;QueueUtils;true;synchronizedQueue;;;Element of Argument[0];Element of ReturnValue;value",
+        "org.apache.commons.collections;QueueUtils;true;transformingQueue;;;Element of Argument[0];Element of ReturnValue;value",
+        "org.apache.commons.collections;QueueUtils;true;unmodifiableQueue;;;Element of Argument[0];Element of ReturnValue;value"
+      ]
+  }
+}
+
+/**
+ * Value-propagating models for the classes `org.apache.commons.collections4.SetUtils`
+ * and `org.apache.commons.collections4.SetUtils$SetView`.
+ */
+private class ApacheSetUtilsModel extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        "org.apache.commons.collections4;SetUtils$SetView;true;copyInto;;;Element of Argument[-1];Element of Argument[0];value",
+        "org.apache.commons.collections4;SetUtils$SetView;true;createIterator;;;Element of Argument[-1];Element of ReturnValue;value",
+        "org.apache.commons.collections4;SetUtils$SetView;true;toSet;;;Element of Argument[-1];Element of Argument[0];value",
+        "org.apache.commons.collections4;SetUtils;true;difference;;;Element of Argument[0];Element of ReturnValue;value",
+        "org.apache.commons.collections4;SetUtils;true;difference;;;Element of Argument[1];Element of ReturnValue;value",
```

Comment:
According to the [documentation](https://commons.apache.org/proper/commons-collections/javadocs/api-4.4/org/apache/commons/collections4/SetUtils.html#difference-java.util.Set-java.util.Set-):

> The returned view contains all elements of a that are not a member of b.

Would elements of `Argument[1]` be in the return value? If not, this row should be removed.